infer/environment.cc

```diff
@@ -1617,6 +1625,7 @@ Environment::processBinding(core::Context ctx, const cfg::CFG &inWhat, cfg::Bind
                             if (auto e = ctx.beginError(bind.loc, core::errors::Infer::CastTypeMismatch)) {
                                 e.setHeader("Argument does not have asserted type `{}`", castType.show(ctx));
                                 e.addErrorSection(ty.explainGot(ctx, ownerLoc));
+                                e.addErrorSections(std::move(errorDetailsCollector));
```

Comment:
Calling `addErrorSections` directly since the error doesn't have an "expected" and "got".